### PR TITLE
'continueOnError' parameter only accepts boolean literals #493

### PR DESCRIPTION
**Pull request by ivanduplenskikh**
- add warning message for variables types checking

Issue: [#493](https://github.com/microsoft/azure-pipelines-vscode/issues/493)

### CODE REVIEW

File: language-service/src/parser/jsonParser.ts [hunk 1]
```diff
@@ -184,8 +184,18 @@ export class ASTNode {
 		}
 		else if (schema.type) {
 			if (this.type !== schema.type) {
+				if (
+					this.type === 'string'
+					&& /^\$({{.*}}|\[.*\]|\(.*\))$/.test(this.getValue())
```
[the remark this answers is not in the document]
Comment (ivanduplenskikh) [reply language-service/src/parser/jsonParser.ts]: Thanks for the advice. Changed to 3 separate checks.

File: language-service/src/parser/jsonParser.ts [hunk 2]
```diff
@@ -184,8 +184,18 @@ export class ASTNode {
...
+				) {
+					validationResult.addProblem({
```
[the remark this answers is not in the document]
Comment (ivanduplenskikh) [reply language-service/src/parser/jsonParser.ts]: Thanks for the advice. Excluded calling the `addProblem` method when the value is an expression

**Review by kirill-ivlev — APPROVED** [review]

File: language-service/src/parser/jsonParser.ts [hunk 3]
```diff
@@ -186,11 +186,23 @@ export class ASTNode {
...
 				if (this.type != "number" || schema.type != "string") {
-					validationResult.addProblem({
-						location: { start: this.start, end: this.end },
-						severity: ProblemSeverity.Warning,
-						getMessage: () => schema.errorMessage || localize('typeMismatchWarning', 'Incorrect type. Expected "{0}".', schema.type as string)
-					});
+					let isVariableExpression = false;
+
+					if (this.type === 'string') {
+						const currentValue = String(this.getValue());
```
Comment (50Wliu) [inline language-service/src/parser/jsonParser.ts:192]: Question - what type does getValue() return? I find the casting to be a bit strange but I also don't have the code in front of me to check at the moment.

Also, can you add a comment in this if-block, saying something like "Ignore expressions as those will be replaced by Azure Pipelines"?

Comment (50Wliu) [reply language-service/src/parser/jsonParser.ts:193]: Ah, quite annoying...now I remember that this code is _mostly_ typed but with awkward `any`s sprinkled around.

**Review by 50Wliu — APPROVED** [review]

**Comment by 50Wliu** [comment]
Excited for this, now I won't have to deal with warning squiggles anymore on my expressions :).

**Comment by ivanduplenskikh** [comment]
> Excited for this, now I won't have to deal with warning squiggles anymore on my expressions :).

Could you please merge this PR, if everything is ok? :)
And our team will update the dependency in [azure-pipelines-vscode](https://github.com/microsoft/azure-pipelines-vscode) repo

**Pull request merged**